src/main/java/net/pms/dlna/DLNAResource.java

```diff
@@ -2892,13 +2916,26 @@ private void addImageResource(StringBuilder sb, DLNAImageResElement resElement)
 		if (!resElement.isResolutionKnown() && DLNAImageProfile.JPEG_RES_H_V.equals(resElement.getProfile())) {
 			throw new IllegalArgumentException("Resolution cannot be unknown for DLNAImageProfile.JPEG_RES_H_V");
 		}
-		String url = DLNAImageProfile.JPEG_RES_H_V.equals(resElement.getProfile()) ?
-			"JPEG_RES" + resElement.getWidth() + "x" + resElement.getHeight() :
-			resElement.getProfile().toString();
-		url = resElement.isThumbnail() ? getThumbnailURL(url) : getURL(url + "_");
+		String url;
+		if (resElement.isThumbnail()) {
+			url = getThumbnailURL(resElement.getProfile());
+		} else {
+			url = getURL(
+				(DLNAImageProfile.JPEG_RES_H_V.equals(resElement.getProfile()) ?
+					"JPEG_RES" + resElement.getWidth() + "x" + resElement.getHeight() :
+					resElement.getProfile().toString()
+				) + "_"
+			);
+		}
 		if (StringUtils.isNotBlank(url)) {
 			String ciFlag;
-			if (resElement.getCiFlag() == null) {
+			/*
+			 * Some Panasonic TV's can't handle if the thumbnails have the CI
+			 * flag set to 0 while the main resource doesn't have a CI flag.
+			 * DLNA dictates that a missing CI flag should be interpreted as
+			 * if it were 0, so the result should be the same.
+			 */
+			if (resElement.getCiFlag() == null || resElement.getCiFlag() == 0) {
```

Comment:
@Nadahar I can confirm that it works but adding extension for thumbnails doesn't solve the problem for me, only this workaround solve it.